### PR TITLE
Added Barbican setup

### PR DESCRIPTION
**Pull request by jgrassler**
This pull request adds the configuration neccessary to get Barbican up and running.

### CODE REVIEW

File: scripts/keystone_data.sh [hunk 1]
```diff
@@ -367,3 +367,20 @@ if [[ "$ENABLED_SERVICES" =~ "magnum-api" ]]; then
...
+    BARBICAN_SERVICE_HOST=$SERVICE_HOST
+    BARBICAN_SERVICE_PORT=${BARBICAN_SERVICE_PORT:-9311}
+
+    create_service_user "barbican" "admin"
+
+    get_or_create_service "barbican" "key-manager" "Barbican Key Manager Service"
+    get_or_create_endpoint \
+        "key-manager" \
+        "RegionOne" \
+        "$BARBICAN_SERVICE_PROTOCOL://$BARBICAN_SERVICE_HOST:$BARBICAN_SERVICE_PORT/v2" \
```
Comment (toabctl) [inline scripts/keystone_data.sh]: Not sure if the ```v2``` at the end is needed. See https://git.openstack.org/cgit/openstack/barbican/tree/devstack/lib/barbican?h=master#n365

Comment (jgrassler) [inline scripts/keystone_data.sh]: Definitely not needed. We figured that one out the hard way this afternoon :-)
(i.e. the endpoint URL must be versionless)

**Comment by jgrassler** [comment]
Do not merge this yet, we'll need to take care of the `[filter:authtoken]` section in `barbican-api-paste.ini`, too, as we discovered today.

**Comment by mjura** [comment]
LGTM

**Comment by toabctl** [comment]
+1 (tested the deployment and a simple ```openstack secret list```

**Comment by jgrassler** [comment]
Sure. Setting this to [wip] again until it's done.

**Comment by jgrassler** [comment]
Ok, all done (rebased/prettified).

File: scripts/openstack-quickstart-demosetup [hunk 2]
```diff
@@ -230,6 +236,12 @@ if [ "x$with_magnum" = "xyes" ]; then
     setup_keystone_authtoken /etc/magnum/magnum.conf magnum keystone_auth
 fi
 
+if [ "x$with_barbican" = "xyes" ]; then
+    setup_keystone_authtoken /etc/barbican/barbican.conf barbican
+    crudini --set /etc/barbican/barbican.conf keystone_authtoken auth_version 'v3.0'
+    sed -i 's/barbican_api/barbican-api-keystone/g' /etc/barbican/barbican-api-paste.ini
```
Comment (toabctl) [inline scripts/openstack-quickstart-demosetup]: Please use crudini and do what devstack is doing: https://github.com/openstack/barbican/blob/master/devstack/lib/barbican#L141

**Comment by jgrassler** [comment]
Note: once this commit is merged, https://github.com/SUSE-Cloud/automation/pull/998 can also be merged (it's got enough +1s already).

**Comment by rsalevsky** [comment]
+1

**Comment by jgrassler** [comment]
Just one thing that occured to me. The `barbican-api` package changed a little from back when we first submitted this, so we'll need an additional `cp barbican-api.conf.sample barbican-api.conf`. I'll quickly fix this...

**Comment by jgrassler** [comment]
Ok, all done. Sorry for the last minute addition (line 644 in `scripts/openstack-quickstart-demosetup`).

**Comment by toabctl** [comment]
Did a basic test (```openstack ca list```) and it works. +1

**Comment by mjura** [comment]
Awesome +1 from too

**Pull request merged**